### PR TITLE
Email Verification Screen /UI

### PR DESCRIPTION
**Pull request by anjalikumari2207**
# Email Verification Screen

This repository contains the code for the Email Verification Screen, a component that verifies a user's email address as part of the sign-up or authentication process.

## Table of Contents

- [Overview](#overview)
- [Features](#features)
- [Components](#components)

## Overview

The Email Verification Screen is a user interface component designed to handle email verification processes in web applications. It typically sends a verification code to the user's email and requires the user to input that code to verify their email address.

## Features

- Sends a verification code to the user's email
- Validates the entered verification code
- Resends verification codes upon request
- Provides user feedback for successful or failed verification attempts
- It has a Style Sheet File of the Email Verification Screens in Screens Folder in the name of the EmailVerification.styles.ts
- And there is a NextButton.tsx Component in UI Folder
    

Screen

### CODE REVIEW

File: src/UI/Screens/EmailVerificationScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+import { View, Text, TextInput, TouchableOpacity, StyleSheet, SafeAreaView, Image } from 'react-native';
+import styles from './Email_Verification.styles'
+
+
+const EmailVerificationScreen = () => {
```
Comment (Jha2508) [inline src/UI/Screens/EmailVerificationScreen.tsx]: function docs

File: src/UI/Screens/EmailVerificationScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        <Text style={styles.title}>Select your personal</Text>
+        <Text style={styles.title}>email ID</Text>
+        <Text style={styles.subtitle} numberOfLines={2}>
+           This email ID will be linked {'\n'}
+          <Text style={styles.iconContainer}>to your account</Text>
+          </Text>
+        <TextInput
+          style={styles.input}
+          placeholder="Select your email ID"
+          placeholderTextColor={'#898A8D'}
```
Comment (Jha2508) [inline src/UI/Screens/EmailVerificationScreen.tsx]: move this color to a colors.ts file and use from there

File: src/UI/Screens/EmailVerificationScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          value={emailID}
+          onChangeText={setEmailID}
+        />
+        <TextInput
+          style={styles.input}
+          placeholder="Enter your name"
+          placeholderTextColor={'#898A8D'}
```
Comment (Jha2508) [inline src/UI/Screens/EmailVerificationScreen.tsx]: move to colors.ts file

File: src/UI/Screens/EmailVerificationScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          value={name}
+          onChangeText={setName}
+          maxLength={40}
+        />
+        <Text style={styles.charCount}>{name.length}/40</Text>
+        <TouchableOpacity style={styles.button}>
+          <Text style={styles.buttonText}>Next</Text>
```
Comment (Jha2508) [inline src/UI/Screens/EmailVerificationScreen.tsx]: Take the button to as a component. Move button to a separate file and use that component over here. that will result in making a component library for our repo

File: src/UI/Screens/EmailVerificationScreen.tsx (hunk 1 above)
Comment (anjalikumari2207) [reply src/UI/Screens/EmailVerificationScreen.tsx]: Done Sir..

File: src/UI/Screens/EmailVerificationScreen.tsx (hunk 2 above)
Comment (anjalikumari2207) [reply src/UI/Screens/EmailVerificationScreen.tsx]: Done Sir..

File: src/UI/Screens/EmailVerificationScreen.tsx (hunk 3 above)
Comment (anjalikumari2207) [reply src/UI/Screens/EmailVerificationScreen.tsx]: Done Sir..

File: src/UI/Screens/EmailVerification.styles.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
+import { StyleSheet } from "react-native";
+
+const styles = StyleSheet.create({
+
+    container: {
+        backgroundColor: '#FFFFE4',
```
Comment (Aashirvaad-P) [inline src/UI/Screens/EmailVerification.styles.ts]: Put these hex codes and other such constants in a constant in file inside config folder for now.

File: src/UI/Screens/EmailVerification.styles.ts
Comment (Aashirvaad-P) [inline src/UI/Screens/EmailVerification.styles.ts:1]: Put these files in a folder by the name of the screen. 
File name for style is .d.ts not .styles.ts

File: src/Assets/color.tsx [hunk 6]
```diff
@@ -0,0 +1,7 @@
+const Colors =
```
Comment (Aashirvaad-P) [inline src/Assets/color.tsx]: This one also should be moved to config folder.

File: src/UI/Components/NextButton.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
```
Comment (Aashirvaad-P) [inline src/UI/Components/NextButton.tsx]: Is there anything special about next button? If not, you can have generic button which accepts text and width or something like that. Most of the styles should be here for the button.

If there is need to give styles at the spot or pass them as parameters then this component would fulfill the purpose of reusable button.

**Comment by Aashirvaad-P** [comment]
Update the branch as well.

File: src/UI/Screens/EmailVerification.styles.ts (hunk 5 above)
Comment (anjalikumari2207) [reply src/UI/Screens/EmailVerification.styles.ts]: Already done sir

File: src/Assets/color.tsx (hunk 6 above)
Comment (anjalikumari2207) [reply src/Assets/color.tsx]: done sir

File: src/UI/Components/Button.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { StyleSheet, Text, View,Alert } from 'react-native';
+import { TouchableOpacity } from 'react-native';
+import styles from "../Screens/EmailVerificationScreen/styles";
+
+const Button = (props:ButtonProps)=>{
```
Comment (Jha2508) [inline src/UI/Components/Button.tsx]: docs

File: src/UI/Components/Button.tsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        <View>
+        <TouchableOpacity onPress={props.onClick} style={styles.button}>
+
+        <Text style={styles.buttonText}>Next</Text>
+        </TouchableOpacity> 
+        </View>       
+    );
+}
+
+type ButtonProps={
```
Comment (Jha2508) [inline src/UI/Components/Button.tsx]: docs

File: src/UI/Screens/EmailVerificationScreen/index.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react';
+import { View, Text, TextInput, TouchableOpacity, StyleSheet, SafeAreaView, Image, Platform, PixelRatio } from 'react-native';
+import styles from './styles';
+import Button from '../../Components/Button';
+import { Pressable } from 'react-native';
+import { Colors } from '../../../Assets/Colors';
+import { Keyboard } from 'react-native';
+import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view"; 
+
+const EmailVerificationScreen = () => {
```
Comment (Jha2508) [inline src/UI/Screens/EmailVerificationScreen/index.tsx]: docs

File: src/UI/Screens/SkipNomineeScreen/index.tsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+          onPress={props.onClose}
+        >
+          <Text style={styles.buttonText}>Close</Text>
+        </TouchableOpacity>
+      </View>
+    </View>
+  );
+};
+
+type SkipNomineeScreenProps = {
```
Comment (Jha2508) [inline src/UI/Screens/SkipNomineeScreen/index.tsx:60]: docs

**Review by Jha2508 — APPROVED** [review]

**Pull request merged**